rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotCommandHandle.py

```diff
@@ -542,78 +540,21 @@ def filter_waypoints(self, wps: list, threshold=1.0):
+        # We assume the first waypoint is safe for pruning if it is
+        # within a threshold of the robot's current position
         first_position = waypoints[0].position
         if len(waypoints) > 2 and\
-                self.dist(first_position, last_pose) > threshold:
-            changed = False
-            index = 0
-            while (not changed):
-                if self.dist(waypoints[index].position, first_position) > 0.1:
-                    changed = True
-                    break
-                waypoints[index].position = last_pose
-                index = index + 1
-
-        if (self.perform_filtering is False):
-            return (first, waypoints)
-
-        changed = False
-        # Find the first waypoint
-        index = 0
-        while (not changed and index < len(waypoints)):
-            if (self.dist(last_pose, waypoints[index].position) < threshold):
-                first = waypoints[index]
-                last_pose = waypoints[index].position
-            else:
-                break
-            index = index + 1
-
-        while (index < len(waypoints)):
-            parent_index = copy.copy(index)
-            wp = waypoints[index]
-            if (self.dist(wp.position, last_pose) >= threshold):
-                changed = False
-                while (not changed):
-                    next_index = index + 1
-                    if (next_index < len(waypoints)):
-                        if (self.dist(waypoints[next_index].position,
-                                      waypoints[index].position) < threshold):
-                            if (next_index == len(waypoints) - 1):
-                                # append last waypoint
-                                changed = True
-                                wp = waypoints[next_index]
-                                wp.approach_lanes =\
-                                    waypoints[parent_index].approach_lanes
-                                second.append(wp)
-                        else:
-                            # append if next waypoint changes
-                            changed = True
-                            wp = waypoints[index]
-                            wp.approach_lanes =\
-                                waypoints[parent_index].approach_lanes
-                            second.append(wp)
-                    else:
-                        # we add the current index to second
-                        changed = True
-                        wp = waypoints[index]
-                        wp.approach_lanes =\
-                            waypoints[parent_index].approach_lanes
-                        second.append(wp)
-                    last_pose = waypoints[index].position
-                    index = next_index
-            else:
-                index = index + 1
+                self.dist(first_position, last_pose) < threshold and\
+                abs(first_position[2] - last_pose[2]) < 0.1:
```

Comment:
What is the rationale for checking this difference in `yaw`?

Reply:
That was initially added to prevent unnecessary spinning in front of doors when the robot gets interrupted, just tested and seems like the spinning only occurs at higher RTF (> 2.0) with or without checking yaw. Removed in last commit